onellm/utils/async_helpers.py

```
@@ -0,0 +1,155 @@
+        loop = asyncio.get_running_loop()
+    except RuntimeError:
+        # No running loop - we can safely create one
+        loop = None
+    
+    if loop is not None:
+        # We're already in an async context
+        # This is a programming error - the caller should use await
+        raise RuntimeError(
+            "Cannot use synchronous method from async context. "
+            "Use the async version (acreate, aupload, etc.) instead."
+        )
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)